docker/credentials_helper.go

```diff
@@ -70,7 +96,59 @@ func GetCredentialsFromCredsStore(registry string) (types.DockerAuthConfig, erro
 		}
 	}
 
-	return result, fmt.Errorf("failed to get credentials from helper specified in ~/.docker/config.json: %w", ErrCredentialsNotFound)
+	return result, fmt.Errorf("failed to get credentials from helper specified in ~/.docker/config.json: %w", errCredentialsNotFound)
+}
+
+func setCredentialsByCredentialHelper(confFilePath, registry, username, secret string) error {
+	helper, err := getCredentialHelperFromConfig(confFilePath)
+
+	if helper == "" || os.IsNotExist(err) {
+		return errNoCredentialHelperConfigured
+	}
+	if err != nil {
+		return fmt.Errorf("failed to get helper from config: %w", err)
+	}
+
+	helperName := fmt.Sprintf("docker-credential-%s", helper)
+	p := client.NewShellProgramFunc(helperName)
+
+	return client.Store(p, &credentials.Credentials{ServerURL: registry, Username: username, Secret: secret})
+}
+
+func listCredentialHelpers() []string {
+	path := os.Getenv("PATH")
+	paths := strings.Split(path, string(os.PathListSeparator))
+
+	helpers := make(map[string]bool)
+	for _, p := range paths {
+		fss, err := ioutil.ReadDir(p)
+		if err != nil {
+			continue
+		}
+		for _, fi := range fss {
+			if fi.IsDir() {
+				continue
+			}
+			if !fi.Mode().IsRegular() {
+				continue
+			}
```

Comment:
Working for me if we remove this, my docker-credentials helper is a symlink and thus it is not being resolved correctly.